### PR TITLE
Add GridAdmin webapp

### PR DESCRIPTION
**Pull request by Tristan-WorkGH**
Add `gridadmin-app` service/pod in deployment:
  - [x] Docker-compose service
  - [x] Kubernetes pod
  - [x] Update documentation

### CODE REVIEW

**Review by etiennehomer — CHANGES_REQUESTED** [review]
Configurations missing : 
- github/workflows/main.yml : `- gridadmin_app_updated`
- all k8s/live/azure-dev and k8s/live/azure-integ config
- k8s/README.md to update
- k8s/live/local/kustomization.yaml

**Review by etiennehomer — CHANGES_REQUESTED** [review]
Missing : 
k8s/live/local/gridstudy-app-idpSettings.json
k8s/live/local/kustomization.yaml

**Comment by Tristan-WorkGH** [comment]
> Missing :
> k8s/live/local/gridstudy-app-idpSettings.json
> k8s/live/local/kustomization.yaml
> k8s/live/azure-dev/patch_tls_ingress.yaml
> docker-compose/study/.env:COMPOSE_PROJECT_NAME=gridadmin

> k8s/live/azure-dev/patch_tls_ingress.yaml
I dont see what to modify, it define the SSL/TLS certificat to use for all gridsuite.

> docker-compose/study/.env:COMPOSE_PROJECT_NAME=gridadmin
No, that the prefix of the compose project when you are in "study" folder, `COMPOSE_PROJECT_NAME` is the name of the folder here.

**Comment by SlimaneAmar** [comment]
Don't forget gridsuite-deployment !

File: docker-compose/study/docker-compose.override.yml [hunk 1]
```diff
@@ -515,3 +515,27 @@ services:
       resources:
         limits:
           memory: 384m
+
+  gridadmin-app:
+    profiles:
+      - all
+      - suite
+      - study
+      #- study-light
```
Comment (antoinebhs) [inline docker-compose/study/docker-compose.override.yml:524]: why keep this commented?

**Comment by etiennehomer** [comment]
I don't know where we should fix it. But we have this log when running gridadmin-app with "npm start" : 
`gridstudy-gateway-1                        | 2024-01-18T10:34:54.215Z  INFO 1 --- [or-http-epoll-3] o.g.g.f.TokenValidatorGlobalPreFilter    : /notify: 401 Unauthorized, https://login.microsoftonline.com/7502dad5-d64c-46c7-9d47-b16f580fcfa9/v2.0 Issuer is not in the issuers white list`

**Review by antoinebhs — APPROVED** [review]
I tested this deployment:
* with local k8s (minikube) instance
* with docker-compose
* I checked with `kustomize build` with this changes and the changes on gridsuite-deployment with live/azure, live/gridsuite-recette and live/gridsuite-dev. It looks ok. We need to check that the deployments are OK once we merge this.

**Comment by antoinebhs** [comment]
Remove Gridadmin on azure deployment

File: k8s/overlays/azure/kustomization.yaml [hunk 2]
```diff
@@ -21,3 +21,10 @@ configMapGenerator:
 patchesStrategicMerge:
 - limit-gridexplore-cases.yaml
 - fluentd-containerd.yaml # default fluentd parse is for docker, on azure we use containerd
+- |-
+  apiVersion: v1
+  kind: Deployment
+  metadata:
+    name: gridadmin-app
+  $patch: delete
+  # We disable/don't deploy GridAdmin in the Azure/demo environment
```
Comment (antoinebhs) [inline k8s/overlays/azure/kustomization.yaml:30]: `Error: accumulating resources: accumulation err='accumulating resources from '../../overlays/azure': '/home/bouhoursant/Devin/gridsuite-deployment/bases/rte/mirror-deployment/k8s/overlays/azure' must resolve to a file': recursed accumulation of path '/home/bouhoursant/Devin/gridsuite-deployment/bases/rte/mirror-deployment/k8s/overlays/azure': no matches for Id Deployment.v1.[noGrp]/gridadmin-app.[noNs]; failed to find unique target for patch Deployment.v1.[noGrp]/gridadmin-app.[noNs]`
on ` kustomize build .`
Have you tested this?

Comment (Tristan-WorkGH) [reply k8s/overlays/azure/kustomization.yaml]: oups, there was a bad `apiVersion: v1` instead of  `app/v1`

**Comment by antoinebhs** [comment]
As discussed with @celmhari, we need the same structure as other apps (griddyna, gridmerge, etc.) for gridadmin as we don't want to deploy it on all the environments:
* a base in k8s/bases/gridadmin
* the associated resource folder in k8s/resources/user-admin (? as admin is already taken) where we can move the resources related to gridadmin-app

**Review by antoinebhs — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]

**Review by etiennehomer — CHANGES_REQUESTED** [review]

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**